presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcWarnings.java

```diff
@@ -122,69 +134,47 @@ public void testStatementWarnings()
 
     @Test
     public void testLongRunningStatement()
-            throws SQLException, InterruptedException
+            throws Exception
     {
         ExecutorService queryExecutor = newSingleThreadExecutor(daemonThreadsNamed("test-%s"));
         QueryCreationFuture queryCreationFuture = new QueryCreationFuture();
         queryExecutor.submit(() -> {
             try {
-                statement.execute("CREATE SCHEMA blackhole.blackhole");
-                statement.execute("CREATE TABLE blackhole.blackhole.test_table AS SELECT 1 AS col1 FROM tpch.sf1.lineitem CROSS JOIN tpch.sf1.lineitem");
+                statement.execute("" +
+                        "CREATE TABLE blackhole.blackhole.test_long_running AS " +
+                        "SELECT * FROM blackhole.blackhole.slow_table");
                 queryCreationFuture.set(null);
             }
             catch (Throwable e) {
                 queryCreationFuture.setException(e);
             }
         });
-        while (statement.getWarnings() == null) {
-            Thread.sleep(100);
-        }
-        SQLWarning warning = statement.getWarnings();
-        Set<WarningEntry> currentWarnings = new HashSet<>();
-        assertTrue(currentWarnings.add(new WarningEntry(warning)));
-        for (int warnings = 1; !queryCreationFuture.isDone() && warnings < 100; warnings++) {
-            for (SQLWarning nextWarning = warning.getNextWarning(); nextWarning == null; nextWarning = warning.getNextWarning()) {
-                // Wait for new warnings
-            }
-            warning = warning.getNextWarning();
-            assertTrue(currentWarnings.add(new WarningEntry(warning)));
-            Thread.sleep(100);
-        }
-        assertEquals(currentWarnings.size(), 100);
+
+        Set<WarningEntry> currentWarnings = getStatementWarnings(statement, queryCreationFuture);
+        assertThat(currentWarnings).size().isGreaterThanOrEqualTo(PRELOADED_WARNINGS + 1);
```

Comment:
`assertThat(currentWarnings).size().isGreaterThan(PRELOADED_WARNINGS);`

Reply:
Previously we were checking there were 100 warnings, not we're checking there are at least 6.
I don't know where this 100 came from, but in any case, it would be best to check actual messages of these warnings.

Reply:
The messages are not useful, as they are generated in `TestingWarningCollector` by rendering the warning code. However, the code and message are compared for uniqueness via the `WarningEntry` wrapper which is then inserted into a `Set`.

The previous test code was confusing and didn't really test what we wanted, which is to ensure that warnings show up as the query is running. I rewrote the code to directly test that.